### PR TITLE
Content POST Changes

### PR DESCRIPTION
**Pull request by khanna98**

### CODE REVIEW

File: controllers/content/content.controller.js [hunk 1]
```diff
@@ -67,6 +78,54 @@ const getContent = async (req, res) => {
...
+
+    const newPostData = {
+      title,
+      content, 
+      type: CONTENT.TYPES.DEFAULT,
+      flag: CONTENT.FLAGS.PENDING,
+      isSpam: CONTENT.SPAM_TYPES.DEFAULT,
+      tags: [],
+      reviews: {
+        isReviewed: false,
```
Comment (sgurjeet) [inline controllers/content/content.controller.js:99]: defaults already set in model. We can remove all field except the ones in body

File: controllers/content/content.controller.js [hunk 2]
```diff
@@ -67,6 +78,54 @@ const getContent = async (req, res) => {
...
+        reviewers: []
+      },
+      sourceMetaData: {},
+      acl: {
```
Comment (sgurjeet) [inline controllers/content/content.controller.js:103]: We can skip this for now.
@haloboy777 will update ACL on verification

File: controllers/content/content.controller.js [hunk 3]
```diff
@@ -67,6 +78,54 @@ const getContent = async (req, res) => {
...
+        updatedBy: {
+          name,
+          email
+        }
+      }
+    }
+
+    const newPostResponse = await ContentModel.create(newPostData);
+
+    return successResponse({res, responseObject: newPostResponse})
```
Comment (sgurjeet) [inline controllers/content/content.controller.js:117]: frontend doesn't really need to created object. Just a success/error response.
We can save payload size here.
+format

File: controllers/content/content.middleware.js [hunk 4]
```diff
@@ -1,55 +1,115 @@
...
+      return {
+        code: STATUS_CODES.NO_CONTENT,
+        message: MESSAGES.CONTENT.ERROR.NO_EMAIL,
+      };
+    } else {
+      if (email && email.length > 0) {
+        //? I was not able to use the email validator, and installed a new one
+        //? We need to see why Regex is taking too long
+        // return emailValidator.validate(email)
+        return validateEmail(email)
```
Comment (sgurjeet) [inline controllers/content/content.middleware.js:84]: let me know if you want to connect to debug this

**Review by haloboy777 — APPROVED** [review]
Looks good to me. I'll add the remaining functionality after this merge.

**Pull request merged**